astropy/io/ascii/core.py

```diff
@@ -642,11 +642,11 @@ def get_cols(self, lines):
         else:
             for i, line in enumerate(self.process_lines(lines)):
                 if i == start_line:
+                    self.names = next(self.splitter([line]))
                     break
             else:  # No header line matching
                 raise ValueError("No header line found in table")
 
-            self.names = next(self.splitter([line]))
```

Comment:
Please revert, this is a code logic change and out of scope for a ruff PR.

Reply:
⚖️ depends on one's perspective. How often do we actually look over these lines of code with a critical eye? In plenty of other ruff PRs we've embraced the opportunity to improve a line of code beyond the minimum level required to satisfy ruff.

Reply:
But if you want to make an Issue and do a follow up PR now that this has been spotted, that's the more procedurally correct thing.
OTOH, the logic elsewhere has often been "great! Someone wants to contribute and is using the ruff PR to also improve code = win/win".